tests/components/withings/test_sensor.py

```diff
@@ -46,10 +29,14 @@ async def test_all_entities(
 ) -> None:
     """Test all entities."""
     await setup_integration(hass, polling_config_entry)
+    entity_registry = er.async_get(hass)
+    entities = er.async_entries_for_config_entry(
+        entity_registry, polling_config_entry.entry_id
+    )
 
-    for sensor in MEASUREMENT_SENSORS + SLEEP_SENSORS:
-        entity_id = await async_get_entity_id(hass, sensor.key, USER_ID, SENSOR_DOMAIN)
-        assert hass.states.get(entity_id) == snapshot
+    for entity in entities:
+        if entity.platform == Platform.SENSOR:
```

Comment:
Would be nice if we could avoid this condition. Tests should preferably only have one branch.

Reply:
But that would mean it tested all entities instead of sensors and I want this test to be pure sensor. What's your opinion on this?

Reply:
I'd patch the `PLATFORMS` constant so that only the sensor platform is set up.

Reply:
I thought that it was a bad practice to do?

Reply:
It's one of two exceptions I can think of where we do patch our code. The other one is timeout constants. Since it's constants it's acceptable I think.